python/ray/tests/conftest.py

```diff
@@ -195,3 +201,23 @@ def two_node_cluster():
     # The code after the yield will run as teardown code.
     ray.shutdown()
     cluster.shutdown()
+
+
+@pytest.fixture()
+def iam_client_stub():
+    boto_config = Config(retries={"max_attempts": BOTO_MAX_RETRIES})
+    resource = RESOURCE_CACHE.setdefault(
```

Comment:
This is here so we don't re-create a un-stubbed client?

Reply:
Right - the stubbed client will be retrieved from the resource cache every time it is requested.